### PR TITLE
Simplify and fix CNN

### PR DESCRIPTION
**Pull request by jlibovicky**
This PR simplifies CNN network for image processing. Normalization layer were removed as well as dropout between convolutional layers. In the spirit of the ongoing refactor, all tensors were rewritten using properties with `@tensor` annotation.

### CODE REVIEW

File: neuralmonkey/encoders/cnn_encoder.py [hunk 1]
```diff
@@ -76,97 +57,103 @@ def __init__(self,
...
-                for i, (filter_size,
-                        n_filters,
-                        pool_size) in enumerate(convolutions):
-                    with tf.variable_scope("cnn_layer_{}".format(i)):
-                        last_layer = conv2d(last_layer, n_filters, filter_size)
+        self._image_height = image_height
+        self._image_width = image_width
+        self._pixel_dim = pixel_dim
+        self._convolutions = convolutions
+        self._fully_connected = fully_connected
```
Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: Nechával bych věci z konfiguráku public, mam pocit, že nedává moc smysl je zakrejvat.

File: neuralmonkey/encoders/cnn_encoder.py [hunk 2]
```diff
@@ -76,97 +57,103 @@ def __init__(self,
...
+
+        self.train_mode = tf.placeholder(tf.bool, shape=[],
```
Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: self train mode mam jako tensor.

File: neuralmonkey/encoders/cnn_encoder.py [hunk 3]
```diff
@@ -76,97 +57,103 @@ def __init__(self,
...
+                                         name="mode_placeholder")
+        self.image_processing_layers = []  # type: List[tf.Tensor]
```
Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: tohle bych dal taky jako tenzor a states pak bude jednoduchá funkcička, ne?

**Comment by jlibovicky** [comment]
@jindrahelcl  ping

File: neuralmonkey/encoders/cnn_encoder.py [hunk 4]
```diff
@@ -76,97 +57,112 @@ def __init__(self,
...
+            s.value for s in self.states.get_shape()[1:]]
+        # pylint: enable=no-member
+        return tf.reshape(
+            self.states,
+            [-1, last_width * last_height, last_n_channels])
+
+    @tensor
     def _attention_mask(self) -> tf.Tensor:
-        return self.__attention_mask
+        return tf.ones(tf.shape(self._attention_tensor)[:2])
```
Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: ones?

Comment (jlibovicky) [inline neuralmonkey/encoders/cnn_encoder.py]: A co jinýho? Prostě chci dělat attention přes celej ten obrázek. Časem bych to vytunil, že image reader bude jetě vracet masku, co si dopadoval a co bylo původně v tom obrázku, ale teď bych to nechal tak, že se prostě pojede přes celý obrázek.

Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: předtim tam byla nějaká chytristika s nulovejma sloupcema nebo co

Comment (jlibovicky) [inline neuralmonkey/encoders/cnn_encoder.py]: byla, ale s ní to nefungovalo

Comment (jindrahelcl) [inline neuralmonkey/encoders/cnn_encoder.py]: Aha, ok :-)
